agent/session_persister.py

```
@@ -0,0 +1,453 @@
+                content=content,
+                tool_name=msg.get("tool_name"),
+                tool_calls=tool_calls_data,
+                tool_call_id=msg.get("tool_call_id"),
+                finish_reason=msg.get("finish_reason"),
+            )
+        except Exception as e:
+            logger.debug("Session DB log_msg failed: %s", e)
+
+    # -- Bulk persistence -----------------------------------------------------
+
+    def persist(self, messages: List[Dict], conversation_history: List[Dict] = None):
```

Ruff (0.15.2)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)